### PR TITLE
Timed lyrics (2)

### LINKED ISSUE #44: is it possible Sync lyrics?
hi, to start, what an awesome project its a really good app!, my question or request is the next one, is it possible to make the lyrics page also capable to show and process sync lyrics. thank you

### PR DESCRIPTION
**Pull request by vixalien**
closes #53 fix #44

### CODE REVIEW

File: src/components/player/lyrics.ts [hunk 1]
```diff
@@ -94,6 +132,171 @@ export class LyricsView extends Gtk.Stack {
...
+    ) {
+      lyrics.timed_lyrics.forEach((line) => {
+        const row = new TimedLyricLineRow(line);
+
+        this._timed_listbox.append(row);
+      });
+
+      this._timed_source.label = lyrics.source;
+
+      this.set_visible_child(this._timed_window);
```
Comment (DaPigGuy) [inline src/components/player/lyrics.ts:153]: This line is in both branches of the `if/else`

File: src/components/player/lyrics.ts [hunk 2]
```diff
@@ -94,6 +132,171 @@ export class LyricsView extends Gtk.Stack {
...
+      this.on_timestamp();
+
+      return GLib.SOURCE_REMOVE;
+    });
+  }
+
+  private on_timestamp() {
+    const player = get_player();
+
+    if (!this.lyrics || !this.lyrics.timed) {
```
Comment (DaPigGuy) [inline src/components/player/lyrics.ts:251]: `!this.lyrics?.timed`

File: src/components/player/lyrics.ts [hunk 3]
```diff
@@ -94,6 +132,171 @@ export class LyricsView extends Gtk.Stack {
...
+
+    const line_id = this.lyrics.timed_lyrics.findIndex((line) => {
+      const timestamp_milli = player.timestamp / 1000;
+      return timestamp_milli >= line.start && timestamp_milli <= line.end;
+    });
+
+    if (line_id < 0) {
+      this._timed_listbox.unselect_all();
+      return;
+    } else {
```
Comment (DaPigGuy) [inline src/components/player/lyrics.ts:263]: Unnecessary `else`

File: src/components/player/lyrics.ts [hunk 4]
```diff
@@ -94,6 +132,171 @@ export class LyricsView extends Gtk.Stack {
...
+      const row = this._timed_listbox.get_row_at_index(line_id);
+
+      if (row) {
```
Comment (DaPigGuy) [inline src/components/player/lyrics.ts:266]: Why not `row && row !== this._timed_listbox.get_selected_row()` to reduced nesting

**Review by DaPigGuy — APPROVED** [review]
Haven't had the chance to test this yet but code LGTM with a few nits

File: src/components/player/lyrics.ts (hunk 1 above)
Comment (vixalien) [reply src/components/player/lyrics.ts:153]: which code??

it's `this.set_visible_child(this._timed_window);` in the if branch and `this.set_visible_child(this._lyrics_window);` in the else branch

File: src/components/player/lyrics.ts (hunk 4 above)
Comment (vixalien) [reply src/components/player/lyrics.ts:266]: that adds another if block.

File: src/components/player/lyrics.ts (hunk 1 above)
Comment (DaPigGuy) [reply src/components/player/lyrics.ts:153]: Whoops

File: src/components/player/lyrics.ts [hunk 5]
```diff
@@ -94,6 +132,171 @@ export class LyricsView extends Gtk.Stack {
...
+        if (row === this._timed_listbox.get_selected_row()) {
+          return;
+        }
```
Comment (DaPigGuy) [inline src/components/player/lyrics.ts]:
```suggestion
      if (row && row !== this._timed_listbox.get_selected_row()) {
```

**Pull request merged**